### ISSUE TITLE
Funding Prompts at the Time of Creation

### ISSUE THREAD

**Issue by stravid87** [globe-and-citizen/Celebrity-Fanalyzer#484]
**Description**
This particular issue is very important to me. If no one else uses Celebrity Fanalyzer or Globe & Citizen -- I personally will use it as I really, really, want to the see the worlds opinion displayed as an anthrogram on many important issues. In order to ensure that people participate and answer / engage with my prompts, I need a way to easily fund it. When I want lots of opinions, I would like to add lots of prize money. If something more simple, a little money is good enough. 

When the winner is chosen, they, obviously, get the money that is held in escrow from the original funding.

CompetitionEscrow Contract Overview
The CompetitionEscrow contract is designed to facilitate competitions where users (competitors) submit entries for a given prompt, and a winner is selected to receive a prize. The contract handles the following key processes:

**User Story**
As a Prompt Creator
I want to poll the world about a topic of interest
When I create a prompt
I see that it is created and can release the funds to users who put entries.

As a registered user
I want to enter a competition to win a crypto prize
When I create an entry and win
I see myself tagged as the winner and the crypto put into my wallet. 

**Acceptance Criteria**
1. Prompt Creation and Fund Deposit: When a new competition prompt is created, the competition organizer (depositor) deposits the prize money into the escrow contract. (The depositor doesn't need to specify the winner at this stage and each deposit generates a unique escrow code that helps in tracking and managing the escrow.)
2. Setting the Winner: After the competition ends, and a winner is selected, the depositor sets the recipient (winner) of the prize money in the contract. The recipient is identified using the unique escrow code generated during the deposit, which links the deposited funds to a specific competition.
3. Releasing the Funds: The winner (recipient) can then release the prize money from the escrow by calling the release function.
The contract ensures that only the recipient, identified by the unique escrow code, can release the funds, ensuring secure and accurate transfer of the prize money.
4. Refunding the Funds: If the competition is canceled or no winner is selected, the depositor can refund the prize money back to themselves by calling the refund function. This ensures that the funds are securely returned to the depositor if the competition does not proceed as planned.
5. Pausing and Unpausing: The contract owner can pause and unpause the contract to prevent or allow these actions as needed (e.g., during maintenance or in case of an issue). Pausing the contract prevents new deposits, setting recipients, releasing, and refunding funds, providing a way to manage contract activity safely.

**Comment by achillegley** [comment]
CompetitionEscrow Contract Overview
The CompetitionEscrow contract is designed to facilitate competitions where users (competitors) submit entries for a given prompt, and a winner is selected to receive a prize. The contract handles the following key processes:

Prompt Creation and Fund Deposit:
When a new competition prompt is created, the competition organizer (depositor) deposits the prize money into the escrow contract.
The depositor doesn't need to specify the winner at this stage.
Each deposit generates a unique escrow code that helps in tracking and managing the escrow.
Setting the Winner:
After the competition ends, and a winner is selected, the depositor sets the recipient (winner) of the prize money in the contract.
The recipient is identified using the unique escrow code generated during the deposit, which links the deposited funds to a specific competition.
Releasing the Funds:
The winner (recipient) can then release the prize money from the escrow by calling the release function.
The contract ensures that only the recipient, identified by the unique escrow code, can release the funds, ensuring secure and accurate transfer of the prize money.
Refunding the Funds:
If the competition is canceled or no winner is selected, the depositor can refund the prize money back to themselves by calling the refund function.
This ensures that the funds are securely returned to the depositor if the competition does not proceed as planned.
Pausing and Unpausing:
The contract owner can pause and unpause the contract to prevent or allow these actions as needed (e.g., during maintenance or in case of an issue).
Pausing the contract prevents new deposits, setting recipients, releasing, and refunding funds, providing a way to manage contract activity safely.

smart-contract source-code: https://github.com/globe-and-citizen/smart-contracts/blob/main/contracts/CompetitionEscrow.sol

**Comment by marclupanc** [comment]
Additional required changes:

```[tasklist]
### Tasks
- [ ] Payment claiming should be available to writers - blocked by #548 
- [ ] Fix - amount 0.00 after amount was sent in transaction details
```
